src/widgets/index/index.ts

```diff
@@ -213,19 +246,23 @@ const index = (props: IndexProps): Index => {
       localWidgets = localWidgets.concat(widgets);
 
       if (localInstantSearchInstance && Boolean(widgets.length)) {
-        helper!.setState(
-          getLocalWidgetsSearchParameters(localWidgets, {
+        privateHelperSetState(helper!, {
+          state: getLocalWidgetsSearchParameters(localWidgets, {
             uiState: localUiState,
             initialSearchParameters: helper!.state,
-          })
-        );
+          }),
+          _isFromAddWidget: true,
+        });
 
         widgets.forEach(widget => {
           if (localInstantSearchInstance && widget.init) {
             widget.init({
               helper: helper!,
               parent: this,
-              uiState: {},
+              // Only index widget uses this key. This means that the initial value
+              // is sufficient, since that index didn't yet exist. If we see wrong
+              // behavior with this, use mainIndex.getWidgetState()
+              uiState: localInstantSearchInstance._initialUiState,
```

Comment:
I've added a test that highlights how global ui state before this change gets "lost". Without this change nested indices don't receive a UiState to start from

Reply:
I got confused because of the comment—I think it's fair to remove it.

Although the name is `uiState`, it has always only acted as an initial UI state from the beginning since it's used to initialize the index UI state.